tests/Doctrine/Tests/ORM/Tools/Pagination/PaginatorTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Tools\Pagination;
+
+use Doctrine\ORM\Decorator\EntityManagerDecorator;
+use Doctrine\ORM\Internal\Hydration\AbstractHydrator;
+use Doctrine\ORM\Query;
+use Doctrine\ORM\Tools\Pagination\Paginator;
+use Doctrine\Tests\Mocks\ConnectionMock;
+use Doctrine\Tests\Mocks\DriverMock;
+use Doctrine\Tests\OrmTestCase;
+
+class PaginatorTest extends OrmTestCase
+{
+    public function testPaginationWithSubSelectStripsParametersInSelect()
+    {
+        $connection = $this->getMockBuilder(ConnectionMock::class)
+            ->setConstructorArgs([[], new DriverMock()])
+            ->setMethods(['executeQuery'])
+            ->getMock()
+        ;
+
+        $em = $this->getMockBuilder(EntityManagerDecorator::class)
+            ->setConstructorArgs([$this->_getTestEntityManager($connection)])
+            ->setMethods(['newHydrator'])
+            ->getMock()
+        ;
+
+        $abstractHydrator = $this->createMock(AbstractHydrator::class);
+        $em->method('newHydrator')->willReturn($abstractHydrator);
+
+        $paramInWhere     = 1;
+        $paramInSubSelect = 2;
+        $returnedIds      = [10];
+
+        $query = new Query($em);
+        $query->setDQL(
+            "SELECT u,
+                (
+                    SELECT MAX(a.version)
+                    FROM Doctrine\\Tests\\Models\\CMS\\CmsArticle a
+                    WHERE a.user = u AND 1 = :paramInSubSelect
+                ) AS HIDDEN max_version
+            FROM Doctrine\\Tests\\Models\\CMS\\CmsUser u
+            WHERE u.id = :paramInWhere"
+        );
+        $query->setParameters(compact('paramInWhere', 'paramInSubSelect'));
```

Comment:
Let's please never ever ever ever use `compact` :-)

Reply:
So @Majkl578 can you add it to forbidden functions in CS standard? I was careful in not introducing anything new violations.

Reply:
I think it is there: https://github.com/doctrine/coding-standard/blob/6432a4e17ef37199794559e86eddb05b519832c0/lib/Doctrine/ruleset.xml#L59

Reply:
Ah it's because 2.6 is locked to version 1

https://github.com/doctrine/doctrine2/blob/f1143f591f1bfcda269141759a20d117e54217b3/composer.json#L30

Any particular reason for that?